driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -71,7 +72,8 @@
                     new GridFSFileCodecProvider(),
                     new Jsr310CodecProvider(),
                     new JsonObjectCodecProvider(),
-                    new BsonCodecProvider()));
+                    new BsonCodecProvider(),
```

Comment:
I kind of feel we should be using the `org.bson.conversions.DEFAULT_CODEC_REGISTRY` as the base and extend it for the local overrides settings.
eg:

```
fromRegistries(
   fromProviders(asList(
       new DBRefCodecProvider(),
       new DBObjectCodecProvider(),
       new DocumentCodecProvider(new DocumentToDBRefTransformer()),
       new IterableCodecProvider(new DocumentToDBRefTransformer()),
       new MapCodecProvider(new DocumentToDBRefTransformer()),
       new GeoJsonCodecProvider(),
       new GridFSFileCodecProvider()
   )),
   org.bson.conversions.DEFAULT_CODEC_REGISTRY
);
```

But determining if that could cause an issue has made my head hurt.

Reply:
Ack - will you open a ticket?

Reply:
Now I see why we didn't do this.  The MongoClientSettings one has a bunch of somewhat subtle alterations because of its support for DBRef.  So I don't see how it could be done straightforwardly.

Reply:
I think that is fine, as the document codec provider with the changes are added after the bson defaults. It does mean there are two DocumentCodecProviders registered - which is a pain.

Reply:
I think we can resolve this as won't fix :)